etna/search/views.py

```diff
@@ -474,9 +474,23 @@ def process_api_result(self, form: Form, api_result: Dict[str, Any]):
                         value.get("sum_other_doc_count", 0)
                     )
 
+    def get_datalayer_data(self, request: HttpRequest) -> Dict[str, Any]:
+        data = super().get_datalayer_data(request)
+
+        data.update(
+            customMetric2=self._custom_metric2,
+        )
+        return data
+
     def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
         context = super().get_context_data(**kwargs)
         context["selected_filters"] = self.get_selected_filters(self.form)
+
+        # sum of all values accross all selected filters
+        self._custom_metric2 = context["selected_filters_count"] = sum(
+            map(len, context["selected_filters"].values())
```

Comment:
Hi Allan,
 I am not sure what is it doing: sum(map(len, context["selected_filters"].values()), can you please explain?

Reply:
Hi Mo,
The explanation is given just above in the comment in simple terms. Furthermore, [map](https://docs.python.org/3.9/library/functions.html#map) calculates the length of each context value (list), then sum adds them up.  Python offers to create neat one liner code like this.

Reply:
So map will apply the length to each "context["selected_filters"].values()" , the resulet will be another iterable transformation after applying the len function, for example [20,40,60,70] , then the sum, will sum the array to 190, which means you will have the selected_filters_count=190 , is that correct?